compiler/x/codegen/SIMDTreeEvaluator.cpp

```diff
@@ -42,7 +42,7 @@ static TR::MemoryReference* ConvertToPatchableMemoryReference(TR::MemoryReferenc
       // Hence, the unresolved memory reference must be evaluated into a register first.
       //
       TR::Register* tempReg = cg->allocateRegister();
-      generateRegMemInstruction(LEARegMem(cg), node, tempReg, mr, cg);
+      generateRegMemInstruction(LEARegMem(), node, tempReg, mr, cg);
```

Comment:
This bug forces us to generate 64-bit `lea` instruction always.

Reply:
I think this bug may have had the side effect of doing the right thing for AOT. Because this code is guarded by an "is unresolved" check, you may want to do
```
generateRegMemInstruction(LEARegMem(cg->comp()->compileRelocatableCode()), node, tempReg, mr, cg);
```

Reply:
Why would AOT or not matter here? The line being modified here is determining whether we should generate a 64-bit or 32-bit load effective address instruction.

Reply:
In AOT, it's always a 64-bit (or I suppose a pointer-sized) load address, because you don't know if the address you want is always going to be 32-bit or 64 bit.

Reply:
I see other code which only seems to do this for 64-bit and always generates 64-bit `lea` there:
https://github.com/eclipse/omr/blob/eeb1035617c5b0beea82d81bfda9eed89e8da743/compiler/x/codegen/FPTreeEvaluator.cpp#L337-L349

Could we get someone more familiar with x86 codegen to chime in? Perhaps @0xdaryl. The code being reviewed here is used for `vload` and `vloadi` evaluators. There is no AOT related code in the above snippet for evaluating `TR::dload` for example, so why should there be one for evaluating `TR::vload`? Also why is there a `is64Bit` check for the `TR::dload` case in FPTreeEvaluator.cpp above?